### PR TITLE
Model Provider Service

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### What is the current behavior?

Everything depends on the UML data model.

Issue Number: #138 

### What is the new behavior?

Model specific information is obtained via a service. Having multiple services allows us to support different data models.

### Does this PR introduce a breaking change?
- [X] Yes
- [ ] No

Tasks:

- [x] populate model selection dropdown from ModelProvider
- [x] Create a model provider that aggregates individual model providers
- [x] Create separate lookups for model elements and diagram items (replace `gaphor/storage/diagramitems.py`)
- [x] Move Sanitizer service to `gaphor.UML`
- [x] `ElementDispatcher` should use new model lookup
- [x] `storage.load` should use new model lookup
- [X] `gaphorconvert,py` should load all model providers - via ModelProvider
- [x] Rename `ModelProvider` to `ModelingLanuage`, as it covers the intent better.

Postpone these tasks, otherwise the PR will grow even more:

- Add namespace filter to ModelProvider
- Should create a ModelProvider for the core model as well? - Not now
- Should Namepace also be part of the core model? (Namespace view depends on it) - not now
- Create elements via `ElementFactory.create()` by name (string). E.g. `element_factory.create("Package")`.
- move property page glade file to UML
- what to do with icons?

### CODE REVIEW

File: gaphor/diagram/propertypages.py [hunk 1]
```diff
@@ -312,6 +311,8 @@ def __init__(self, subject: NamedElement):
         self.watcher = subject.watcher() if subject else DummyEventWatcher()
 
     def construct(self):
+        from gaphor import UML
```
Comment (danyeaw) [inline gaphor/diagram/propertypages.py:314]: Do we need a comment here explaining why this import can't be at the top of the module?

File: gaphor/services/modelprovider.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+class ModelProviderService(Service, ModelProvider):
+
+    DEFAULT_PROFILE = "UML"
+
+    def __init__(self, properties={}):
+        """
+        Create a new Model Provider. It will provide all models defined
+        as entrypoints under `[gaphor.modelproviders]`.
+
+        Te `properties` argument is optional, in which case the service will
```
Comment (danyeaw) [inline gaphor/services/modelprovider.py:16]: Typo

**Comment by danyeaw** [comment]
This looks great, you are on a roll! :rocket:

File: gaphor/diagram/propertypages.py (hunk 1 above)
Comment (amolenaar) [reply gaphor/diagram/propertypages.py:314]: Maybe it needs to move to the UML package as well. I'm not sure yet.

**Comment by amolenaar** [comment]
Thanks. It's really nice to see when a plan comes together. It all falls into place :)

File: gaphor/diagram/propertypages.py (hunk 1 above)
Comment (amolenaar) [reply gaphor/diagram/propertypages.py]: I moved the NamedElement editor to the UML package.

**Comment by amolenaar** [comment]
@danyeaw Something like this? :)

File: gaphor/SysML/modelinglanguage.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+"""
+The UML Model Provider is the entrypoint for an application to
+UML related assets.
```
Comment (danyeaw) [inline gaphor/SysML/modelinglanguage.py:3]: Should we update this docstring for SysML?

**Comment by danyeaw** [comment]
@amolenaar This look really great! :tada: 

I did notice when doing some brief operational testing that I was getting some TypeErrors. For example, put a Class on a diagram and then click off of it so it is no longer selected:
`TypeError: unsubscribe_all() takes 0 positional arguments but 1 was given`

File: gaphor/SysML/modelinglanguage.py (hunk 3 above)
Comment (amolenaar) [reply gaphor/SysML/modelinglanguage.py]: Done.

**Comment by amolenaar** [comment]
I fixed the `TypeError`.

**Comment by danyeaw** [comment]
@amolenaar Really nice job! :shipit: 🎉

**Comment by amolenaar** [comment]
Thanks 🤗

**Pull request merged**